pkg/config/datafileprojectconfig/mappers/condition_trees.go

```diff
@@ -101,7 +101,7 @@ func buildConditionTree(conditions interface{}) (conditionTree *entities.TreeNod
 			n := &entities.TreeNode{}
 			if err := createLeafCondition(v, n); err != nil {
 				retErr = err
-				return
+				return nil, nil, retErr
```

Comment:
Probably because I don't read Go much, but would this be more clear:
```go
if err := createLeafCondition(v, n); err != nil {
    conditionTree = nil 
    odpSegments = nil    
    retErr = err    
    return conditionTree, odpSegments, retErr
```
Similar on line 47...but then again, allocating those variables is a waste. Bah. This is probably something familiar to Go devs

Reply:
I think, in this error case, it should be fine to send nil directly as the error is the main priority here. but thanks for the thought.